pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -47,19 +49,24 @@ const (
 
 type LaunchTemplateProvider struct {
 	sync.Mutex
+	logger                *zap.SugaredLogger
 	ec2api                ec2iface.EC2API
 	amiProvider           *AMIProvider
 	securityGroupProvider *SecurityGroupProvider
 	cache                 *cache.Cache
 }
 
-func NewLaunchTemplateProvider(ec2api ec2iface.EC2API, amiProvider *AMIProvider, securityGroupProvider *SecurityGroupProvider) *LaunchTemplateProvider {
-	return &LaunchTemplateProvider{
+func NewLaunchTemplateProvider(ctx context.Context, ec2api ec2iface.EC2API, amiProvider *AMIProvider, securityGroupProvider *SecurityGroupProvider) *LaunchTemplateProvider {
+	l := &LaunchTemplateProvider{
 		ec2api:                ec2api,
+		logger:                logging.FromContext(ctx).Named("launchtemplate"),
 		amiProvider:           amiProvider,
 		securityGroupProvider: securityGroupProvider,
 		cache:                 cache.New(CacheTTL, CacheCleanupInterval),
 	}
+	l.cache.OnEvicted(l.onCacheEvicted)
+	l.hydrateCache(ctx)
```

Comment:
This ctx won't benefit from the annotated logger. You need to do something like `ctx = ...` on line 60.

Reply:
Oh perhaps this doesn't matter.

Reply:
yeah, it's not async so I was just going to use the ctx passed for initialization